src/mesh/coordinates.cxx

```diff
@@ -170,6 +170,120 @@ Coordinates::Coordinates(Mesh *mesh)
   }
 }
 
+Field2D interpolate_and_Neumann(const Field2D &f, CELL_LOC location, bool diagonal) {
+  Mesh* localmesh = f.getMesh();
+  Field2D result = interp_to(f, location, RGN_NOBNDRY);
+  localmesh->communicate(result);
+
+  // Copy nearest value into boundaries so that differential geometry terms can
+  // be interpolated if necessary
+  // Note: cannot use applyBoundary("neumann") here because applyBoundary()
+  // would try to create a new Coordinates object since we have not finished
+  // initializing yet, leading to an infinite recursion
+  for (auto bndry : localmesh->getBoundaries()) {
+    if (bndry->bx != 0) {
+      for(bndry->first(); !bndry->isDone(); bndry->next1d()) {
+        for (int i=0; i<localmesh->xstart; i++)
```

Comment:
This just sets values at one end, do you need to set them at the other end too?

Reply:
No, the other end is a different `bndry`, and bndry->bx/bndry-by change sign so that the index goes in the opposite direction. Maybe I should add some comments...